src/main/core/slave.c

```diff
@@ -338,12 +338,12 @@ static gchar** _slave_generateEnvv(Slave* slave, InterposeMethod interposeMethod
         g_free(timestring);
     }
 
-    if (interposeMethod == INTERPOSE_PRELOAD_ONLY) {
-        envv = g_environ_setenv(envv, "SHADOW_INTERPOSE_METHOD", "PRELOAD_ONLY", 0);
-    } else if (interposeMethod == INTERPOSE_PTRACE_ONLY) {
-        envv = g_environ_setenv(envv, "SHADOW_INTERPOSE_METHOD", "PTRACE_ONLY", 0);
-    } else if (interposeMethod == INTERPOSE_PRELOAD_PTRACE) {
-        envv = g_environ_setenv(envv, "SHADOW_INTERPOSE_METHOD", "PRELOAD_PTRACE", 0);
+    if (interposeMethod == INTERPOSE_PRELOAD) {
+        envv = g_environ_setenv(envv, "SHADOW_INTERPOSE_METHOD", "PRELOAD", 0);
```

Comment:
codecov thinking this line is uncovered seems concerning. Could be a bug in our coverage analysis, but could you double-check that the `preload` test cases are being generated and are actually using preload-mode (e.g. by checking the shadow log output)?

Could there be a bug in the `add_shadow_tests` macro that's not actually generating all the test cases?

Reply:
TL;DR You're right, the preload tests are not running in coverage mode because we have [configured them not to run](https://github.com/shadow/shadow/blob/cefa1d2cfcfefc88dd86d249dc5fcf879b63c29d/ci/container_scripts/test.sh#L7) due to #867. I found no indication of errors introduced in the PR or in the recently merged test macros.

----

There are 174 total tests that run on my local machine, including 35 preload mode tests.

```
$ cat build/Testing/Temporary/LastTest.log | grep "interpose-method" | grep arg | grep "hybrid" | wc -l
42
$ cat build/Testing/Temporary/LastTest.log | grep "interpose-method" | grep arg | grep "ptrace" | wc -l
44
$ cat build/Testing/Temporary/LastTest.log | grep "interpose-method" | grep arg | grep "preload" | wc -l
35
```

I checked several CTest files, e.g., `build/src/test/socket/listen/CTestTestfile.cmake` and it appears that all of the preload tests are generated correctly.

I also spot checked some of the Github action logs for the non-coverage tests that passed, and they all appear to be running preload tests too.

The coverage container test only ran 139 total tests, none of which are preload mode. This agrees with my above stats, where indeed the 35 preload tests are missing (174-35=139).

Reply:
Oops, right :man_facepalming: . Sorry about that!